### PR TITLE
Mn 2016 11 notify moderators on idea added2

### PR DESCRIPTION
**Pull request by MagdaN**
Notify moderators without extra field in action table

### CODE REVIEW

File: euth/actions/signals.py [hunk 1]
```diff
@@ -1,18 +1,75 @@
...
+    actor = instance.creator
+
+    action = Action(
+        actor=actor,
+        verb=verb,
+        action_object_content_type=content_type,
+        action_object_object_id=instance.pk,
+    )
+
+    if str(content_type) == 'Comment':
```
Comment (slomo) [inline euth/actions/signals.py]: Comments do also have a project property. So they probably don't need any special treatment. when getting the project:

```
if hasattr(instance, 'project'):
    project = instance.project
else:
   project = None
```

Should be sufficient for all Actionables

File: euth/actions/signals.py [hunk 2]
```diff
@@ -1,18 +1,75 @@
...
+
+def add_action(sender, instance, created, **kwargs):
+    content_type = ContentType.objects.get_for_model(instance)
+    verb = verbs.CREATE if created else verbs.UPDATE
+    actor = instance.creator
+
+    action = Action(
+        actor=actor,
+        verb=verb,
+        action_object_content_type=content_type,
```
Comment (slomo) [inline euth/actions/signals.py]: `action_object =  instance` should to the same here

File: euth/actions/signals.py [hunk 3]
```diff
@@ -1,18 +1,75 @@
...
+    elif isinstance(instance, Item):
+        project = instance.module.project
+        project_contenttype = ContentType.objects.get_for_model(project)
+
+        action.target_content_type = project_contenttype
+        action.target_object_id = project.pk
+        action.project = project
+        action.save()
+
+    else:
```
Comment (slomo) [inline euth/actions/signals.py]: Empty else clause can probably be left out.

File: euth/actions/signals.py [hunk 4]
```diff
@@ -1,18 +1,75 @@
...
+def notify_creator(instance):
+    if hasattr(instance.target, 'creator'):
+        creator = instance.target.creator
+        if creator.get_notifications and not creator == instance.actor:
+            emails.notify_users_on_create_action(instance, [creator.email])
+
+
+def notify_moderators(instance):
+    if str(instance.target_content_type) == 'project':
+        recipients = instance.target.moderators.all().values_list(
```
Comment (slomo) [inline euth/actions/signals.py]: Should the actor be filtered out from this list?

File: euth/actions/signals.py [hunk 5]
```diff
@@ -1,18 +1,75 @@
...
 
+def notify_creator(instance):
+    if hasattr(instance.target, 'creator'):
+        creator = instance.target.creator
+        if creator.get_notifications and not creator == instance.actor:
+            emails.notify_users_on_create_action(instance, [creator.email])
+
+
+def notify_moderators(instance):
+    if str(instance.target_content_type) == 'project':
```
Comment (slomo) [inline euth/actions/signals.py]: In my personal opinion `instance.target_content_type.model_class() is Project` would be preferable, but that is certainly only a question of style.

File: euth/actions/verbs.py [hunk 6]
```diff
@@ -1,7 +1,7 @@
 # http://activitystrea.ms/registry/verbs/
 
-CREATE = 'create'
-UPDATE = 'update'
+CREATE = 'created'
```
Comment (MagdaN) [inline euth/actions/verbs.py]: I thought it should read like a sentence, but I don't mind.

File: euth/actions/signals.py (hunk 5 above)
Comment (MagdaN) [inline euth/actions/signals.py]: Go for it

File: euth/actions/signals.py (hunk 4 above)
Comment (MagdaN) [inline euth/actions/signals.py]: true

File: euth/actions/signals.py (hunk 1 above)
Comment (MagdaN) [inline euth/actions/signals.py]: ok

**Comment by MagdaN** [comment]
@slomo feel free to change

**Pull request merged**